### PR TITLE
[fix](estimate) func call with not filter will estimate some statistics

### PR DESCRIPTION
**Pull request by amorynan**
before this pr: use ! or not in or condition when table has been analyzed we will meet 
```
SELECT
  count(1)
FROM
  table_30_un_pa_ke_pr_di4
where
  col_int_undef_signed_not_null < -128
  or not array_contains(col_array_bigint__undef_signed, col_int_undef_signed_not_null);

ERROR 1105 (HY000): errCode = 2, detailMessage = Not-predicate meet unexpected child: 
  array_contains(col_array_bigint__undef_signed, cast(col_int_undef_signed_not_null as BIGINT))
```

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run external

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java [hunk 1]
```diff
@@ -478,6 +478,7 @@ private Statistics handleUnknownCase(EstimationContext context) {
         return context.statistics;
     }
 
+    // where not array_contains([xx, xx], xx)
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java:481]: move this comment to L511

**Comment by morrySnow** [comment]
replace screenshot with text because we could not add image into git message

**Review by englefly — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java (hunk 1 above)
Comment (amorynan) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java]: done!

**Comment by amorynan** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by amorynan** [comment]
run feut

**Comment by amorynan** [comment]
run external

**Comment by amorynan** [comment]
run p0

**Review by englefly — APPROVED** [review]

**Pull request merged**